### ISSUE TITLE
My Patterns: the pagination doesn't keep the state after click on a pattern  

### ISSUE THREAD

**Issue by miksansegundo** [WordPress/gutenberg#59003]
### Description

The pagination doesn't keep the state in My Patterns page.

I expected that after I use the pagination to see a page, I can click on a pattern, to preview it and then click the `<` back button on the sidebar of the pattern detail to come back to the same page.

### Step-by-step reproduction instructions

- Go to My patterns in the Site Editor
- Use the navigation to see another page
- Click on a pattern
- Click on the `<` back button on the pattern detail page
- Verify the pagination is reset and you see the first page

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/1881481/6370c2a1-820c-42ad-8c87-14a270cc6941

### Environment info

Gutenberg: v17.7.0-rc.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by miksansegundo** [comment]
This same issue affects any Data View options and filters.

I expected those to be also kept when I preview, or even edit a pattern, and I go back to the My patterns via the `<` back button.

|Filters|View Options|
|-|-|
|<img width="285" alt="Screenshot 2567-02-14 at 17 24 13" src="https://github.com/WordPress/gutenberg/assets/1881481/2dc27dcd-a077-47bd-b879-1ff4bb3f3f71">|<img width="195" alt="Screenshot 2567-02-14 at 17 18 36" src="https://github.com/WordPress/gutenberg/assets/1881481/ed4edb02-9fef-4ac3-acb1-31e3da854957">|

**Comment by annezazu** [comment]
cc @ntsekouras

**Comment by ntsekouras** [comment]
Yeah, right now that's not possible. There are some discussions about adding some of the params in the URL that would resolve this. We do this for the layout type right now only(`grid, table`, etc..). There are some nuances with the custom views exploration that could come from database for example and in that case we have two sources of truth, but will probably prioritize the url query params.

**Comment by colorful-tones** [comment]
> will probably prioritize the url query params.

This sounds like a good plan. We’ll need URL query params soon. 👍

**Comment by annezazu** [comment]
Removing this from the 6.5 board since there's not an immediate solution to implement as shown above and [due to the voting that happened in the async triage session this week.](https://wordpress.slack.com/archives/C02QB2JS7/p1708972400219879?thread_ts=1708972393.927249&cid=C02QB2JS7)

**Comment by youknowriad** [comment]
While this is not fully solved at 100%, most of it is.

 - browser back works 
 - there's no pattern details page
 - all dataviews except search and page are persisted
